yt_idefix/_io/vtk_io.py

```diff
@@ -205,7 +205,7 @@ def read_grid_coordinates(
 
 
 def read_field_offset_index(
-    fh: BinaryIO, shape: Shape, *, upper_case_varnames: bool
+    fh: BinaryIO, shape: Shape, *, default_field_list: map
```

Comment:
Thanks to this type hint, I see that my last comments mislead you, let's correct this.
A `map` is a generator (a one-time-only iterator), and `in` doesn't work as you'd think:
```python
>>> it = map(lambda x : 1, range(5))
>>> 1 in it
False
>>> list(it)
[]
```

Because we need to iterate over this more than once within the `while` loop, we need to convert it to a list beforehand
```suggestion
    fh: BinaryIO, shape: Shape, *, default_field_list: list[str]
```

Reply:
Ah I have tested it by
```python
>>> field, info = unzip(ds.field_info.known_other_fields)
>>> "RHO" in field
True
```
Then I thought it should work. I should try to run it again :)